### ISSUE TITLE
Build and publish a release against current Scala 2.12 milestone

### ISSUE THREAD

**Issue by JoshRosen** [scalanlp/breeze#520]
It would be great if Breeze could build and publish artifacts against the current Scala 2.12 milestone (currently 2.12.0-M4). This is a blocker for building Apache Spark against Scala 2.12 (see [SPARK-14438](https://issues.apache.org/jira/browse/SPARK-14438)).

**Comment by dlwh** [comment]
I'm swamped with work for the next 2 weeks so I won't be able to do this
quickly. I'd be happy to help people with issues if they wanted to give it
a shot

**Comment by JoshRosen** [comment]
It looks like we're blocked by a few other libraries which still need to cross-publish, so I'll use this list to track them:

- [ ] macro paradise, which will likely be published soon: https://gitter.im/scalamacros/paradise?at=5704cc58dc7a73775697af7f
- [ ] shapeless
- [ ] spire-math

**Comment by dlwh** [comment]
superseded by #604
